### PR TITLE
Implement platform_get_locale_date_format for Linux

### LINKED ISSUE #5548: platform_get_locale_date_format is not implemented for Linux.
`platform_get_locale_date_format` is not implemented for Linux.

I did try to write an implementation using `std::locale` and `date_order`, but I could not get it to pick up the system locale or anything from the `LANG` or `LC_*` environment variables without throwing runtime exceptions.

### PR DESCRIPTION
**Pull request by SijmenSchoon**
Fixes #5548.

Note that currently, this isn't implemented in the C++ standard library. This means that date_order() will always return no_order, causing this function to always return the default `DATE_FORMAT_DAY_MONTH_YEAR`, as is the case before this commit too.

Once `date_order()` will be properly implemented in the standard library someday, it will however start returning the right order for the current locale, causing this function to start working as intended.

### CODE REVIEW

**Comment by Broxzier** [comment]
The issue mentions that it causes runtime errors. Do you know if that's still the case, or will this not happen with this implementation?

**Comment by SijmenSchoon** [comment]
I have not experienced them myself on an Arch Linux system, and haven't been able to reproduce them in a minimal test of the same code.

I did not do anything with LC_*, which IntelOrca did — `std::locale` picks up the right locale automatically.

When I did put `LC_TIME` in the `std::locale` constructor I did get a segfault, which I assume is what IntelOrca did, but that isn't the correct way.

TL;DR no it will not happen

**Comment by Gymnasiast** [comment]
LGTM. Could you amend the changelog?

**Comment by SijmenSchoon** [comment]
Why did that fail?

**Comment by Gymnasiast** [comment]
Won't work, probably. Discord-rpc frequently borks up.
